### PR TITLE
Ethantr/feature/unavailability form testing

### PR DESCRIPTION
**Pull request by ethantr**
## Describe your changes
Unit tests for the unavailability form view model have been made. 

1. The unit tests check to see if the fields (title, date fields) can be updated and match their input.
2. It checks to see if an exception is raised if the end date is before the start date.
3. It checks if the form is "valid" (all fields filled) logic is correct.
4. It also checks if dispose works.

Next: data layer - repository and client

## Issue ticket number and link
FE-181

### CODE REVIEW

File: fireapp/test/presentation/unavailability_form/unavailability_form_view_model_test.dart [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+import 'package:fireapp/presentation/unavailability_form/unavailability_form_view_model.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+
+import 'unavailability_form_view_model_test.mocks.dart';
+
+// Generate Mocks
+@GenerateMocks([AuthenticationRepository, UnavailabilityFormRepository])
```
Comment (emilymclean) [inline fireapp/test/presentation/unavailability_form/unavailability_form_view_model_test.dart:13]: I typically put this above the import statement for said mocks (i.e, put it above line 13)

**Review by emilymclean — CHANGES_REQUESTED** [review]
Great work, just one minor change and I'll approve :)

File: fireapp/test/presentation/unavailability_form/unavailability_form_view_model_test.dart (hunk 1 above)
Comment (ethantr) [reply fireapp/test/presentation/unavailability_form/unavailability_form_view_model_test.dart:13]: is that what you mean?

Comment (emilymclean) [reply fireapp/test/presentation/unavailability_form/unavailability_form_view_model_test.dart:13]: Like directly above

```
@GenerateMocks([AuthenticationRepository, UnavailabilityFormRepository])
import 'unavailability_form_view_model_test.mocks.dart';
```

**Review by emilymclean — APPROVED** [review]

**Pull request merged**